gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -134,7 +134,30 @@ private <T> T withRevWalkUnchecked(CheckedFunction1<RevWalk, T> fun) {
   }
 
   private boolean isBranchPresent(String branchFullName) {
-    return Try.of(() -> jgitRepoForMainGitDir.resolve(branchFullName)).getOrNull() != null;
+    // If '/' chars exist in the branch name, then this loop is needed in order to avoid possible IDE errors, which
+    // could appear if two files like in the example below exist under .git directory for the repository:
+    // .git/refs/heads/docs/improve_readme (short branch name is `docs/improve_readme` - exists locally)
+    // .git/refs/remotes/upstream/docs (short branch name is `docs` - this branch exists only on upstream, not locally)
+    // In such scenario `org.eclipse.jgit.lib.Repository#resolve` called to check if `docs/improve_readme` branch exists
+    // under .git/refs/remotes/upstream/ will try to find the branch using the following file path:
+    // .git/refs/remotes/upstream/docs/improve_readme
+    // which will end in a "Not a directory" `java.nio.file.FileSystemException` (as existing file
+    // .git/refs/remotes/upstream/docs is not a directory). And that will produce a LOG.error which will generate
+    // an IDE error. So, the cause of this while loop is to avoid such IDE errors.
+    val segments = List.of(branchFullName.split("/"));
+    int numOfSegmentsToUse = 3; // 3 is the least number that can contain the branch name (for `refs/heads/<branch_name>`)
+    boolean result = false;
+    while (!result && numOfSegmentsToUse <= segments.size()) {
+      val testedSegment = segments.take(numOfSegmentsToUse).mkString("/");
+      result = Try.of(() -> jgitRepoForMainGitDir.resolve(testedSegment)).getOrNull() != null;
+      numOfSegmentsToUse++;
+    }
```

Comment:
How about sth like
```suggestion
     // 3 is the least number that can contain the branch name (for `refs/heads/<branch_name>`)
    for (int numOfSegmentsToUse = 3; numOfSegmentsToUse < segments.size(); numOfSegmentsToUse++) {
      val testedPrefix = segments.take(numOfSegmentsToUse).mkString("/");
      if (Try.of(() -> jgitRepoForMainGitDir.resolve(testedSegment)).getOrNull() != null)
        return false;
    }
    
```

Reply:
> prematurely breaking the loop and returning something for the whole method in the middle of the loop.

That's a perfectly valid idiom, esp. in Java :) See e.g. `io.vavr.collection.LinearSeq#indexWhere`, just first example that I've found